kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/metrics/ScanReportSuite.scala

```diff
@@ -0,0 +1,481 @@
+        path,
+        expectException = false,
+        expectedNumAddFiles = 2,
+        expectedNumAddFilesFromDeltaFiles = 2,
+        expectedNumActiveAddFiles = 2,
+        filter = Some(partFilter),
+        expectedPartitionPredicate = Some(partFilter)
+      )
+
+      // With data filter
+      checkScanReport(
+        path,
+        expectException = false,
+        expectedNumAddFiles = 2,
+        expectedNumAddFilesFromDeltaFiles = 2,
+        expectedNumActiveAddFiles = 2,
+        filter = Some(dataFilter),
+        expectedDataSkippingFilter = Some(expectedSkippingFilter)
+      )
+
+      // With data and partition filter
+      checkScanReport(
+        path,
+        expectException = false,
+        expectedNumAddFiles = 2,
+        expectedNumAddFilesFromDeltaFiles = 2,
+        expectedNumActiveAddFiles = 2,
+        filter = Some(new Predicate("AND", partFilter, dataFilter)),
+        expectedDataSkippingFilter = Some(expectedSkippingFilter),
+        expectedPartitionPredicate = Some(partFilter)
+      )
+    }
+  }
+
+  test("ScanReport: close scan file iterator early") {
+    withTempDir { tempDir =>
+      val path = tempDir.getCanonicalPath
+
+      // Set up delta table with 2 add files
+      spark.range(10).write.format("delta").mode("append").save(path)
+      spark.range(10).write.format("delta").mode("append").save(path)
+
+      checkScanReport(
+        path,
+        expectException = false,
+        expectedNumAddFiles = 1,
+        expectedNumAddFilesFromDeltaFiles = 1,
+        expectedNumActiveAddFiles = 1,
+        expectedIsFullyConsumed = false,
+        consumeScanFiles = iter => iter.close() // Close iterator before consuming any scan files
+      )
+    }
+  }
+
+  //////////////////
+  // Error cases ///
+  //////////////////
+
+  test("ScanReport error case - unrecognized partition filter") {
+    // Thrown during partition pruning when the expression handler cannot evaluate the filter
+    // Because partition pruning happens within a `map` on the iterator, this is caught and reported
+    // within `map`
+    withTempDir { tempDir =>
+      val path = tempDir.getCanonicalPath
+
+      // Set up partitioned table
+      spark.range(10).withColumn("part", col("id") % 2)
+        .write.format("delta").partitionBy("part").save(path)
+
+      val partFilter = new Predicate("foo", new Column("part"), Literal.ofLong(0))
+
+      checkScanReport(
+        path,
+        expectException = true,
+        expectedNumAddFiles = 0,
+        expectedNumAddFilesFromDeltaFiles = 0,
+        expectedNumActiveAddFiles = 0,
+        expectedIsFullyConsumed = false,
+        filter = Some(partFilter),
+        expectedPartitionPredicate = Some(partFilter)
+      )
+    }
+  }
+
+  test("ScanReport error case - error reading the log files") {
+    withTempDir { tempDir =>
+      val path = tempDir.getCanonicalPath
+
+      // We set up table with a giberish json file that falls before a file containing the P&M
```

Comment:
How about `We set up a table with a giberish json file at version 0 and a valid json file at version 1 that contains the P&M. .... ... ...`